pinot-tools/src/main/java/org/apache/pinot/tools/tuner/driver/TunerDriver.java

```diff
@@ -0,0 +1,202 @@
+import org.apache.pinot.tools.tuner.strategy.Strategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ *TunerDriver is an executable interface, has three pluggable modules:
+ *   MetaData Manager: a manger for MetaManager, which is an interface to access segment metadata.
+ *   QuerySrc: an iterator interface over input source, has a pluggable AbstractQueryParser, who parses each item in input source, and returns AbstractQueryStats, a wrapper of relevant fields input.
+ *   Strategy, which has four user defined functions operating on a map of Map<Long, Map<String, Map<String, AbstractMergerObj>>>:
+ *                                                                             |		       |				    |				     |
+ *                                                                        ThreadID	   TableName	  ColumnName		  Abstract object of stats for a column
+ *       Filter: A function to filter AbstractQueryStats, by table name, number of entries scanned in filters, number of entries scanned post filter, etc. The relevant AbstractQueryStats will be feed to Accumulator.
+ *       Accumulator: A function to process AbstractQueryStats and MetaManager; then accumulate stats to corresponding AbstractMergerObj entry.
+ *       Merger: A function to merge two AbstractMergerObj entries having the same TableName/ColumnName from different threads.
+ *       Reporter: A function to postprocess and print(email) out the final results of a table.
+ */
+public class TunerDriver {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TunerDriver.class);
+  public static final int NO_CONCURRENCY = 0;
+
+  private QuerySrc _querySrc = null;
+  private MetaManager _metaManager = null;
+  private Strategy _strategy = null;
+  private int _threadPoolSize = 0;
+
+  /**
+   * Set the number of threads used in action
+   * @param threadPoolSize The number of threads used in action
+   * @return this
+   */
+  public TunerDriver setThreadPoolSize(int threadPoolSize) {
+    _threadPoolSize = threadPoolSize;
+    return this;
+  }
+
+  /**
+   * Set the query source, e.g. LogFileSrcImpl
+   * @param querySrc E.g. LogFileSrcImpl
+   * @return this
+   */
+  public TunerDriver setQuerySrc(QuerySrc querySrc) {
+    _querySrc = querySrc;
+    return this;
+  }
+
+  /**
+   * Set the metaManager for caching and query cardinality e.g. MetaManager
+   * @param metaManager E.g. MetaManager
+   * @return this
+   */
+  public TunerDriver setMetaManager(MetaManager metaManager) {
+    _metaManager = metaManager;
+    return this;
+  }
+
+  /**
+   * Set the strategy for the recommendation, e.g. FrequencyImpl, OLSAnalysisImpl, ParserBasedImpl
+   * @param strategy E.g. ParserBasedImpl
+   * @return this
+   */
+  public TunerDriver setStrategy(Strategy strategy) {
+    _strategy = strategy;
+    return this;
+  }
+
+  private Map<Long, Map<String, Map<String, AbstractAccumulator>>> _threadAccumulator = null;
```

Comment:
multiple threads are accessing the same Map. Don't you need a ConcurrentHashMap?